index.bs

```diff
@@ -8292,16 +8292,20 @@ The [=remote end steps=] given |session| and |command parameters| are:
 1. Let |arguments| be the <code>arguments</code> field of |command
    parameters| if present, or an empty [=/list=] otherwise.
 
-1. Let the |contexts| be the value of the <code>contexts</code> field of |command parameters| if it is present or null if it isn’t.
+1. If the <code>contexts</code> field of |command parameters| is present:
 
-1. If |contexts| is not null:
+   1. Let |contexts| be an empty [=/set=].
 
-   1. For each |context id| of |contexts|:
+   1. For each |context id| of |command parameters|["<code>contexts</code>"]
 
       1. Let |context| be the result of [=trying=] to [=get a browsing context=] with |context id|.
 
       1. If |context| is not a [=top-level browsing context=], return [=error=] with [=error code=] [=invalid argument=].
 
+      1. Append |context| to |contexts|.
+
+1. Otherwise let |contexts| to null.
```

Comment:
I assume this is a mistake?

```suggestion
1. Otherwise set |contexts| to null.
```

Reply:
It was not mistake as we had may or may not have had it created before. 
Did not like it so changed the structure to:
1. Set to null
2. Check if param exist 
2.1. Set to Empty Set
2.2. Loop logic

Reply:
This got refactored and doesn't exist anymore.